azure/scope/managedmachinepool.go

```diff
@@ -154,6 +154,11 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		replicas = *machinePool.Spec.Replicas
 	}
 
+	subnetName := managedControlPlane.Spec.VirtualNetwork.Subnet.Name
```

Comment:
are we planning on deprecating this field at all? or is this a fallback/override situation where `managedControlPlane.Spec.VirtualNetwork.Subnet.Name` defined the default for all node pools that can be optionally overridden by each `managedMachinePool.Spec.SubnetName`? Assuming it's the latter but just want to double check

Reply:
yes you are right the managedMachinePool.Spec.SubnetName should be optionally overridden if the user wants to place a specific agentpool in a different subnet. If managedMachinePool.Spec.SubnetName is undefined capz will use managedControlPlane.Spec.VirtualNetwork.Subnet.Name to place the agent-pool.